src/background/modules/CssAnalysis.js

```diff
@@ -3,77 +3,47 @@ import * as BrowserCommunication from "/common/modules/BrowserCommunication/Brow
 import { COMMUNICATION_MESSAGE_TYPE } from "/common/modules/data/BrowserCommunicationTypes.js";
 
 const TAB_FILTER_URLS = ["<all_urls>"];
-let lastSettingsInjection = null;
+let lastContentScript = null;
 
 AddonSettings.setCaching(false);
 
-/**
- * Triggers the new setting in case it has been changed.
- *
- * @public
- * @returns {Promise}
- */
-export async function triggerNewColorStatus() {
-    // add new code injection and remove old one
-    if (lastSettingsInjection) {
-        (await lastSettingsInjection).unregister();
-    }
-    lastSettingsInjection = await enableSettingInjection();
-}
-
 /**
  * Adds a content script that is injected by Firefox and provides the preloaded
  * value of fakedColorStatus.
  *
  * @private
+ * @param {string} [fakedColorStatus]
  * @returns {Promise}
  */
-async function enableSettingInjection() {
-    const functionalMode = await AddonSettings.get("functionalMode");
-    const fakedColorStatus = await AddonSettings.get("fakedColorStatus");
+export async function injectContentScript(fakedColorStatus) {
+    const settings = {
+        fakedColorStatus: fakedColorStatus || await AddonSettings.get("fakedColorStatus"),
+        functionalMode: await AddonSettings.get("functionalMode")
+    };
 
-    return browser.contentScripts.register({
+    // add new code injection
+    const newContentScript = await browser.contentScripts.register({
         matches: TAB_FILTER_URLS,
         js: [{
-            code: `
-                    ${setSettings.toString()}
-                    setSettings(${functionalMode}, "${fakedColorStatus.toUpperCase()}");
+            code: `;
```

Comment:
```suggestion
            code: `
```

That is unnecessary.

Reply:
It is the ghost of a very obscure IE bug. Probably from IE7 era.
You're right, it is not necessary in Firefox.
It has no performance impact, but if you want, I'll remove it.